packages/flutter/lib/src/widgets/change_notifier_builder.dart

```diff
@@ -0,0 +1,151 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'basic.dart';
+import 'framework.dart';
+
+/// Builds a [Widget] when given a concrete value of a [ChangeNotifier].
+///
+/// If the `child` parameter provided to the [ChangeNotifierBuilder] is not
+/// null, the same `child` widget is passed back to this
+/// [ChangeNotifierWidgetBuilder] and should typically be incorporated in the
+/// returned widget tree.
+///
+/// See also:
+///
+///  * [ChangeNotifierBuilder], a widget which invokes this builder each time
+///    a [ChangeNotifier] changes value.
+typedef ChangeNotifierWidgetBuilder<T> = Widget Function(BuildContext context, Widget? child);
+
+/// A widget whose content stays synced with a [ChangeNotifier].
+///
+/// Given a [ChangeNotifier] and a [builder] which builds widgets from
+/// subtypes of `ChangeNotifier`, this class will automatically register itself
```

Comment:
> a [builder] which builds widgets from subtypes of `ChangeNotifier`,

This sounds odd. A ChangeNotifier is not a widget. And the builder should be building any widget it wants, really.